Feature/database/add mentors model

mohmedeprahem opened the pull request.

mohmedeprahem committed:
Add mentor model
- create mentor schema
- add mentor validation messages
- add relation between mentor and reviews
- add relation between mentor and courses
- add relation between mentor and subscriptions

mohmedeprahem committed:
Add some comments in models/mentors.js

mohmedeprahem committed:
Update mentor model
- Edit error message of gender property

mohmedeprahem committed:
Update mentor model
- add match in picture property mentor shema

SamGoher commented on models/mentors.js:
```diff
@@ -0,0 +1,133 @@
+        required: [true, 'Please enter phone number'],
+        match: [/^\d{10}$/, 'Invalide phone number']
+    },
+    countryCode: {
```
please add validation for country code

mohmedeprahem committed:
Update mentor model
- edit validation in mentor Schema

mohmedeprahem committed:
Update mentor model
- edit errror messages validation in mentor schema
- Update min length and max length in some properties validation
- update schema

mohmedeprahem committed:
Update error message validation from ur to your 3l4an tnbast
